packages/common/src/Adaptor.js

```diff
@@ -542,6 +543,92 @@ export function chunk(array, chunkSize) {
   return output;
 }
 
+/**
+ * The function `parseCsv` takes a CSV file string or stream and parsing options as input, and returns a promise that
+ * resolves to the parsed CSV data.
+ * @param {String | Stream} streamOrString - A CSV string or a readable stream
+ * @param {Object} [parsingOptions] - Optional. Parsing options for converting CSV to JSON.
+ *     Possible options:
+ *     - `delimiter` {string|Buffer|[string|Buffer]} - Defines the character(s) used to delimitate the fields inside a record. Default: `','`
+ *     - `quote` {string|Buffer|[string|Buffer]} - Defines the characters used to surround a field. Default: `'"'`
+ *     - `escape` {Buffer|string|null|boolean} - Set the escape character as one character/byte only. Default: `"`
+ *     - `columns` {boolean | array | function} - Generates record in the form of object literals. Default: `true`
+ *     - `bom` {boolean} - Strips the {@link https://en.wikipedia.org/wiki/Byte_order_mark byte order mark (BOM)} from the input string or buffer. Default: `true`
+ *     - `trim` {boolean} - Ignore whitespace characters immediately around the `delimiter`. Default: `true`
+ *     - `ltrim` {boolean} - Ignore whitespace characters from the left side of a CSV field. Default: `true`
+ *     - `rtrim` {boolean} - Ignore whitespace characters from the right side of a CSV field. Default: `true`
+ *     - `chunkSize` {number} - The size of each chunk of CSV data. Default: `Infinity`
+ *     - `skip_empty_lines` {boolean} - The `skip_empty_lines` skips any line which is empty. Default: `true`
+ * @returns {Promise} The function returns a Promise that resolves to the result of parsing a CSV `stringOrStream`.
```

Comment:
The callback argument is not documented. It needs to specify clearly exactly WHEN it is called, and with what.